hashintel/hash
# Enable `@typescript-eslint/no-unnecessary-condition`

## 🌟 What is the purpose of this PR?

When reviewing #899, I noticed [one case of unnecessary condition spotted by LGTM](https://lgtm.com/projects/g/hashintel/hash/rev/pr-e05bdd412ffc66afab6228f325748588d5efad5d). It turned out [a corresponding ESLint rule](https://typescript-eslint.io/rules/no-unnecessary-condition/) was not enabled. This PR fixes this issue and helps devs get earlier feedback on the code they write.




## 🔗 Related links





- [Asana task](https://app.asana.com/0/1202621375435414/1202736501900834/f) _(internal)_

## 🔍 What does this change?

- [`@typescript-eslint/no-unnecessary-condition`](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md) is introduced throughout the codebase

## ⚠️ Known issues

- This PR may cause crashes short term because some ‘unnecessary’ conditions may be ‘necessary’. If this happens, we need to fix the typings first and only then re-introduce the conditions.

- I added `eslint-ignore` in a couple of places, mainly because of `await` in loops (see https://github.com/typescript-eslint/typescript-eslint/issues/1459#issuecomment-614174852)

## 🐾 Next steps

- Fix bugs if any

## 🛡 What tests cover this?

- Playwright
- Manual
- https://github.com/blockprotocol/blockprotocol/pull/505

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Run HASH and click around
1. Check Preview deployment in https://github.com/blockprotocol/blockprotocol/pull/505
